doc/releases/changelog-dev.md

```diff
@@ -391,6 +391,8 @@ same information.
 * Fixed `Identity.__repr__` to return correct wires list.
   [(#6506)](https://github.com/PennyLaneAI/pennylane/pull/6506)
 
+* Fixed a shape error in the .matrix() conversion of products of ops with a pauli_rep, occuring when the parameters are tensor-like.
```

Comment:
Since this bug was not present in the current stable version of PL, I would actually add this PR as a follow up to the entry for the previous PR.